pybamm/models/full_battery_models/lithium_ion/electrode_soh.py

```diff
@@ -31,189 +32,138 @@ class ElectrodeSOH(pybamm.BaseModel):
+            self.initial_conditions[x_100] = pybamm.Scalar(0.9)
 
+        # These variables are defined in all cases
         self.variables = {
-            "C": C,
-            "Capacity [A.h]": C,
             "x_100": x_100,
             "y_100": y_100,
-            "x_0": x_0,
-            "y_0": y_0,
             "Un(x_100)": Un_100,
             "Up(y_100)": Up_100,
-            "Un(x_0)": Un_0,
-            "Up(y_0)": Up_0,
-            "Up(y_0) - Un(x_0)": Up_0 - Un_0,
             "Up(y_100) - Un(x_100)": Up_100 - Un_100,
-            "n_Li_100": 3600 / param.F * (y_100 * Cp + x_100 * Cn),
-            "n_Li_0": 3600 / param.F * (y_0 * Cp + x_0 * Cn),
-            "n_Li": n_Li,
-            "C_n": Cn,
-            "C_p": Cp,
-            "C_n * (x_100 - x_0)": Cn * (x_100 - x_0),
-            "C_p * (y_100 - y_0)": Cp * (y_0 - y_100),
+            "Q_Li": Q_Li,
+            "n_Li": Q_Li * 3600 / param.F,
+            "Q_n": Q_n,
+            "Q_p": Q_p,
         }
 
-    @property
-    def default_solver(self):
-        # Use AlgebraicSolver as CasadiAlgebraicSolver gives unnecessary warnings
-        return pybamm.AlgebraicSolver()
-
-
-class ElectrodeSOHx100(pybamm.BaseModel):
-    """Model to calculate electrode-specific SOH for x_100 and y_100, from [1]_.
-    This model is mainly for internal use, to calculate summary variables in a
-    simulation.
-
-    .. math::
-        n_{Li} = \\frac{3600}{F}(y_{100}C_p + x_{100}C_n),
-    .. math::
-        V_{max} = U_p(y_{100}) - U_n(x_{100}),
-
-    **Extends:** :class:`pybamm.BaseModel`
-    """
-
-    def __init__(self, name="ElectrodeSOHx100 model", param=None):
-        pybamm.citations.register("Mohtat2019")
-        super().__init__(name)
-
-        param = param or pybamm.LithiumIonParameters()
-
-        Un = param.n.prim.U_dimensional
-        Up = param.p.prim.U_dimensional
-        T_ref = param.T_ref
-
-        n_Li = pybamm.InputParameter("n_Li")
-        V_max = pybamm.InputParameter("V_max")
-        Cn = pybamm.InputParameter("C_n")
-        Cp = pybamm.InputParameter("C_p")
-
-        x_100 = pybamm.Variable("x_100")
-        y_100 = (n_Li * param.F / 3600 - x_100 * Cn) / Cp
-
-        Un_100 = Un(x_100, T_ref)
-        Up_100 = Up(y_100, T_ref)
-
-        self.algebraic = {x_100: Up_100 - Un_100 - V_max}
-
-        self.initial_conditions = {x_100: pybamm.Scalar(0.9)}
-
-        self.variables = {"x_100": x_100, "y_100": y_100}
+        # Define variables and equations for 0% state of charge
+        if "x_0" in solve_for:
+            if known_value == "cyclable lithium capacity":
+                x_0 = pybamm.Variable("x_0")
+                Q = Q_n * (x_100 - x_0)
+            elif known_value == "cell capacity":
+                x_0 = x_100 - Q / Q_n
+                Q_Li = y_100 * Q_p + x_0 * Q_n
+            y_0 = y_100 + Q / Q_p
+            Un_0 = Un(x_0, T_ref)
+            Up_0 = Up(y_0, T_ref)
+            if known_value == "cyclable lithium capacity":
```

Comment:
can these just be set in the if statement above?

Reply:
yes, good catch